### DISCUSSION TITLE
Custom attached property

### DISCUSSION THREAD

**Discussion by limeniye** [unoplatform/uno#7748] [Q&A]
I just wanna create attached property. 

In the UWP I created attached property but this attached property does't work for NET6 pratfroms. 
My xamlGenerator has issue for another platforms. 

I readed Row attached property realization from Grid and also I Readed documentation. Its should be like next code:
```C#
[GeneratedDependencyProperty(DefaultValue = 0.0d, AttachedBackingFieldOwner = typeof(UIElement), Attached = true)]
public static readonly DependencyProperty LeftProperty = CreateLeftProperty();

public static double GetLeft(DependencyObject obj)
    => GetLeftValue(obj);

public static void SetLeft(DependencyObject obj, double value)
    => SetLeftValue(obj, value);
```

But **GeneratedDependencyPropertyAttribute** it is **internal sealed class**. 

How can I create my custom attached property?

**Comment by limeniye** [comment]
The simple Example:
```C#
    public static class ExampleDataAp
    {
        public static readonly DependencyProperty DataProperty = 
            DependencyProperty.RegisterAttached("Data", typeof(FrameworkElement), typeof(ExampleDataAp),
                     new PropertyMetadata(false, OnDataChanged));

        private static void OnDataChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
        {
        }

        public static void SetData(UIElement element, FrameworkElement value)
        {
            element.SetValue(DataProperty, value);
        }
        public static FrameworkElement GetData(UIElement element)
        {
            return (FrameworkElement)element.GetValue(DataProperty);
        }
    }
```
```xml
            <ItemsControl.ItemContainerStyle>
                <Style TargetType="ContentPresenter">
                    <Setter Property="dragAPs:ExampleDataAp.Data" Value="{Binding ElementName=baseListView}"/>
                </Style>
            </ItemsControl.ItemContainerStyle>
```

Error
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1024	Preprocessor directive expected	DragPositionUnoProject.Skia.Wpf
```

**Reply by jeromelaban** [reply]
Thanks for the report! Can you create a simple repro project that may show the error?

Also `GeneratedDependencyPropertyAttribute` is an internal class that is not meant to be used outside of Uno itself, as it is not a WinUI API.

**Reply by limeniye** [reply]
Sure.

**Reply by limeniye** [reply]
@jeromelaban, the [repository](https://github.com/limeniye/UnoAttachedPropertyIssueDemonstrateRepository) and [problems](https://github.com/limeniye/UnoAttachedPropertyIssueDemonstrateRepository/blob/main/AttachedProperiesWeirdWorkDemonstrate/AttachedProperiesWeirdWorkDemonstrate.Shared/MainPage.xaml#L42-L49)

**Reply by jeromelaban** [reply]
Thanks. This is a known issue https://github.com/unoplatform/uno/issues/4826

**Reply by limeniye** [reply]
hm. If I understood correctly Uno doesn't support attached property?

**Reply by jeromelaban** [reply]
Uno does support attached properties properly, just not the setter syntax with a binding you're using in your sample.
